tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -616,12 +705,12 @@ private[scalajs] final class ScalaJSClassEmitter(
               }))
 
           case _ =>
-            js.Return(!(!({
-              genIsScalaJSObject(obj) &&
-              ((obj DOT "$classData" DOT "arrayDepth") === depth) &&
-              genIsClassNameInAncestors(className,
-                  obj DOT "$classData" DOT "arrayBase" DOT "ancestors")
-            })))
+            val tagVar = genLet(js.Ident("tag"), mutable = false, obj && (obj DOT "$typeTag"))
+            val tag = tagVar.ref
+            val test = js.BinaryOp(JSBinaryOp.<, tag, js.IntLiteral(0)) &&
+              (((tag >> 23) & 255) === depth) &&
+              genIntervalsTest(className, tag & 8388607)
+            js.Block(tagVar, js.Return(!(!test)))
```

Comment:
Is `!!` necessary? It seems all the components of `test` are always booleans, aren't they?